chalicelib/app_utils.py

```diff
@@ -521,11 +558,10 @@ def process_view_result(connection, res, is_admin):
                 # not yet run, display an icon status to signify this
                 res['assc_action_status'] = 'ready'
 
-            # If we got an action, set its name to 'latest action'
-            # so we can grab it's history via same method as this check
-            latest_action = action.get_latest_result()
-            if latest_action:
-                res['latest_action'] = latest_action.get('name')
+            # This used to try to get the latest result and only populate 'latest_action' if one exists.
+            # Doing so makes the main page take 2-3x as long to load, so we won't be doing that anymore.
+            res['action_history'] = res.get('action')  # = action name
```

Comment:
This is a weird naming convention, it's called `action_history` because this is used to generate the action history link (see base.html). Before this change it was called `latest_action`, which is even more confusing IMO.